### PR TITLE
Api end

### PR DESCRIPTION
**Pull request by altontonn**
# Blog app: add API endpoints 💎 

In this PR, we worked on the following:-

- Created an API endpoint to list all posts for a user.

- Created an API endpoint to list all comments for a user's post.

- Created an API endpoint to add a comment to a post. Remember that the comment owner is the user that makes it; it should not be set by passing a parameter.

- Had API endpoints receive JSON and respond to JSON as well.

### Instructions for API Endpoint usage

1. In the URL add `http://localhost:3000/users/:id.json`
2. In the URL add `http://localhost:3000/users/:id/posts.json`

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+  end
+
   def create
     @comment = Comment.new(author_id: params[:user_id], post_id: params[:post_id], text: params[:comment][:text])
-    if @comment.save
-      redirect_to user_post_path(params[:user_id], params[:post_id]), notice: 'Comment was successfully created.'
-    else
-      render :new
+    redirect_to user_post_path(params[:user_id], params[:post_id]), notice: 'Comment was successfully created.'
+    respond_to do |format|
```
Comment (Qoosim) [inline app/controllers/comments_controller.rb]: - [ ] I got this error while trying to add comment to a post. Kindly look into this.

### Error message

commet-error

### Project Requirements
create-comment

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,10 +1,28 @@
 class CommentsController < ApplicationController
+  def index
+    user = User.find(params[:user_id])
+    post = user.posts.find(params[:post_id])
+    comments = post.comments
+    render json: comments
+  end
```
Comment (Qoosim) [inline app/controllers/comments_controller.rb]: - [ ] The api endpoint for listing all the comments for a post is not working at the moment. Kindly fix it.

### Error
comments-for-a-post

### Project Requirements
list-comment

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -2,6 +2,10 @@ class PostsController < ApplicationController
   def index
     @user = User.find_by(id: params[:user_id])
     @posts = @user.posts.includes(:comments).paginate(page: params[:page], per_page: 2)
+    respond_to do |format|
+      format.html # index.html.erb
+      format.json { render json: @posts }
+    end
   end
```
Comment (Qoosim) [inline app/controllers/posts_controller.rb:9]: - Greate this works as expected. :heavy_check_mark: 

### Result
posts-for-a-user

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM       , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯 
- [x] API endpoints correctly setup🥇 
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**